libraries/chain/block_header.cpp

```diff
@@ -25,6 +25,12 @@ namespace eosio { namespace chain {
       return result;
    }
 
+   bool block_header::is_proper_svnn_block() const {
+      auto exts = validate_and_extract_header_extensions();
```

Comment:
I worry about doing this every time this function is called, since it isn't cached.

I think this should only do the faster `schedule_version == proper_svnn_schedule_version` check. It can be assumed that if that is the case, then the finality extension must also be present. Though that should be validated when validating the block header along with the other special block header validation rules for Proper IF Blocks such as the confirmed count being zero and the proposer signature being computed on the block ID.